### PR TITLE
Unify and enhance block validation

### PR DESCRIPTION
**Pull request by mgdelacroix**
#### Summary
This PR unifies all block validations in an `IsValid` receiver method and adds enhances the logic. 

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-53234

### CODE REVIEW

File: server/services/store/sqlstore/blocks.go [hunk 1]
```diff
@@ -20,18 +20,6 @@ const (
 	descClause     = " DESC "
 )
 
-type ErrEmptyBoardID struct{}
-
-func (re ErrEmptyBoardID) Error() string {
-	return "boardID is empty"
-}
```
Comment (mgdelacroix) [inline server/services/store/sqlstore/blocks.go:27]: `ErrEmptyBoardID` has been moved to the model to match the rest of the errors

File: server/services/store/sqlstore/blocks.go [hunk 2]
```diff
@@ -20,18 +20,6 @@ const (
...
-
-type ErrLimitExceeded struct{ max int }
-
-func (le ErrLimitExceeded) Error() string {
-	return fmt.Sprintf("limit exceeded (max=%d)", le.max)
-}
-
```
Comment (mgdelacroix) [inline server/services/store/sqlstore/blocks.go:34]: This error was unused

**Review by esarafianou — APPROVED** [review]
LGTM, thanks!

**Review by sbishel — APPROVED** [review]

**Pull request merged**